src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,45 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final int DEFAULT_STACK_SIZE = 10;
+    private T[] stackArray = (T[]) new Object[DEFAULT_STACK_SIZE];
```

Comment:
```suggestion
    private T[] stackArray;
```
there is a constructor for this